src/Symfony/Component/Translation/Dumper/XliffFileDumper.php

```diff
@@ -198,6 +198,6 @@ private function dumpXliff2(string $defaultLocale, MessageCatalogue $messages, ?
 
     private function hasMetadataArrayInfo(string $key, array $metadata = null): bool
     {
-        return null !== $metadata && \array_key_exists($key, $metadata) && ($metadata[$key] instanceof \Traversable || \is_array($metadata[$key]));
+        return null !== $metadata && \array_key_exists($key, $metadata) && is_iterable($metadata[$key]);
```

Comment:
```suggestion
        return is_iterable($metadata[$key] ?? null);
```

Reply:
Nice! I've done this change too. Thanks.